content/blog/rune-frequency-202212.md

```diff
@@ -0,0 +1,253 @@
+| `~_` | 103 | 0.2192% | 97.47% |
+| `~%` | 84 | 0.1787% | 97.65% |
+| `?+` | 82 | 0.1745% | 97.82% |
+| `;;` | 72 | 0.1532% | 97.97% |
+| `:^` | 61 | 0.1298% | 98.1% |
+| `\|^` | 58 | 0.1234% | 98.23% |
+| `~&` | 56 | 0.1192% | 98.35% |
+| `~+` | 56 | 0.1192% | 98.47% |
+| `?@` | 55 | 0.117% | 98.58% |
+| `%^` | 53 | 0.1128% | 98.7% |
+| `\|_` | 52 | 0.1106% | 98.81% |
+| `=,` | 45 | 0.09575% | 98.9% |
+| `%_` | 43 | 0.0915% | 98.99% |
+| `^?` | 42 | 0.08937% | 99.08% |
+| `^.` | 39 | 0.08299% | 99.17% |
+| `\|~` | 36 | 0.0766% | 99.24% |
+| `!<` | 35 | 0.07447% | 99.32% |
+| `%.` | 33 | 0.07022% | 99.39% |
+| `\|$` | 31 | 0.06596% | 99.45% |
+| `\|@` | 29 | 0.06171% | 99.51% |
+| `?<` | 26 | 0.05532% | 99.57% |
+| `.*` | 25 | 0.0532% | 99.62% |
+| `=;` | 24 | 0.05107% | 99.67% |
+| `=:` | 24 | 0.05107% | 99.73% |
+| `~?` | 19 | 0.04043% | 99.77% |
+| `!,` | 16 | 0.03405% | 99.8% |
+| `;/` | 16 | 0.03405% | 99.83% |
+| `^\|` | 16 | 0.03405% | 99.87% |
+| `%*` | 12 | 0.02553% | 99.89% |
+| `\|:` | 11 | 0.02341% | 99.92% |
+| `~!` | 8 | 0.01702% | 99.93% |
+| `.^` | 8 | 0.01702% | 99.95% |
+| `!?` | 4 | 0.008511% | 99.96% |
+| `=~` | 4 | 0.008511% | 99.97% |
+| `.?` | 4 | 0.008511% | 99.98% |
+| `^~` | 3 | 0.006384% | 99.98% |
+| `!=` | 2 | 0.004256% | 99.99% |
+| `?#` | 2 | 0.004256% | 99.99% |
+| `;=` | 2 | 0.004256% | 100.0% |
+| `~<` | 1 | 0.002128% | 100.0% |
+| `\|?` | 1 | 0.002128% | 100.0% |
+| `!@` | 0 | 0.0% | 100.0% |
+| `~=` | 0 | 0.0% | 100.0% |
+| `~$` | 0 | 0.0% | 100.0% |
+| `;<` | 0 | 0.0% | 100.0% |
+| `^&` | 0 | 0.0% | 100.0% |
+
+Rune frequency follows a [power-law distribution](https://en.wikipedia.org/wiki/Power_law) with 18 runes representing 80% of total rune utilization.  Indeed, several runes are _hapax legomena_ in `/sys` ([`|?` barwut](https://developers.urbit.org/reference/hoon/rune/bar#-barwut), [`~<`](https://developers.urbit.org/reference/hoon/rune/sig#-siggal)) or do not occur at all (discussed below).
+
+![**Figure 1**.  Observed rune frequency in `/sys` as a power law.](https://storage.googleapis.com/media.urbit.org/blog/rune-frequency.png)
+
+**Figure 1**.  Observed rune frequency in `/sys` as a power law.
+
+(Bear with me on that graph:  it's rather hard to display order runes in any sort of coherent way I've come up with so far.)
+
+
+
+##  Analysis
+
+There aren't any real surprises here.  The most frequent runes reflect the most common design patterns:
+
+- `%` cen rune calls tend to route through [`%:` cencol](https://developers.urbit.org/reference/hoon/rune/cen#-cencol) since the irregular form `(fun 1 2)` desugars to `%:`.
+- [`:*` coltar](https://developers.urbit.org/reference/hoon/rune/col#-coltar) serves similarly as the desugaring of lists constructed by `~[1 2 3]`.
+- [`%=` centis](https://developers.urbit.org/reference/hoon/rune/cen#-centis) is invoked through the irregular `$()` expression resets.  These are most commonly employed in gates and traps as a recursion.
```

Comment:
this is used very extensively for modifying legs as well as recursion, maybe good to mention both

Reply:
eg `this(thing new-thing)` in agents and everywhere with engine patterns